zaza/model.py

```diff
@@ -323,7 +323,8 @@ def get_unit_from_name(unit_name, model=None, model_name=None):
     :rtype: int
     :raises: ServiceNotRunning
     """
-    cmd = "stat -c %Y /proc/$(pidof -x {} | cut -f1 -d ' ')".format(service)
+    cmd = ("stat -c %Y /proc/$(pgrep {} --nslist pid"
```

Comment:
This change is not directly connected to you adding a service restart test for the keystone charm. Please split out into separate PR or at least a separate commit.

Reply:
@fnordahl please see https://github.com/openstack-charmers/zaza/pull/145